### ISSUE TITLE
VS code IntelliSense cannot find Mesh class

### ISSUE THREAD

**Issue by duchaoyu** [compas-dev/compas#767]
Hi, brgcode. @lottilotte and I are working on CSD2 and we both find that VScode Intellisense cannot find the Mesh class (in my win VScode and her mac VScode). When we execute the code, the code works. However, when we write the code, VScode doesn't show any hints. 

```
from compas.datastructures import Network
from compas.datastructures import Mesh
print(Network)
print(Mesh)
```

For example, the VS code can find the Network class and hint the class methods. 
![image](https://user-images.githubusercontent.com/43376817/107221194-97490500-6a13-11eb-9642-de2d486f414f.png)
![image](https://user-images.githubusercontent.com/43376817/107221241-a92aa800-6a13-11eb-8f8b-f002e6ecbc28.png)

However, this Mesh class cannot be found. VS Code only finds the mesh module, but not the class. So, no class method is shown.
![image](https://user-images.githubusercontent.com/43376817/107221490-0161aa00-6a14-11eb-8384-dd8ae079aaf2.png)
![image](https://user-images.githubusercontent.com/43376817/107221645-31a94880-6a14-11eb-8d18-4a337caff807.png)

This is not really a bug or a problem, just a bit annoying for teaching. We don't know whether it's a problem in our VS code, or in COMPAS...  

What I have tried:
1. changing the name of files under `compas.datastructures.mesh`, which doesn't help. 
2. specifying the path. Finally, VScode shows the hint, but I guess this is not the best solution. 
![image](https://user-images.githubusercontent.com/43376817/107222462-3e7a6c00-6a15-11eb-81da-fa09fa2106d5.png)

**Comment by tomvanmele** [comment]
which language server are you using?

if `PyLance` then this is indeed a known problem, which will go away by switching back to, for example, `jedi`

**Comment by duchaoyu** [comment]
![image](https://user-images.githubusercontent.com/43376817/107226444-6fa96b00-6a1a-11eb-987b-ec7678e422a6.png)
Thank you for your reply, Tom. I'm sorry that I don't know language server. Is it in the VScode settings?
If it is, then my language server was `Microsoft`... I changed it to `Jedi` and restarted VScode. VScode somehow changed the language server back to `Microsoft`... but now I see hints in my code.... 
![image](https://user-images.githubusercontent.com/43376817/107226754-cdd64e00-6a1a-11eb-8b2f-a5e00b5734b8.png)
However, this method doesn't work on @lottilotte 's mac....

**Comment by lottilotte** [comment]
Hi,

It says that I’m using Jedi. Or is this not the right place to check?



Many thanks!

**Comment by tomvanmele** [comment]
these are the global settings. it can be that you have a local setting that makes VS Code use a different server. check in `.vscode`

**Comment by lottilotte** [comment]
Hi Tom,

So in my .vscode settings I added the Jedi but it still doesn’t work… I also tried for example the maxxi workspace and also there it’s the same problem. 

I’m sorry to annoy with this. Some of our students of CSD2 are facing the same issue. 

Any other suggestions of what to try? Or could we have a quick zoom call where I can share screen? Many thanks in advance!

Best,
Lotte

**Comment by yck011522** [comment]
i have one dirty hack that Tom is not going to like it:
ignore compas' two level import system. and import Mesh class from the actual document. 

```
# this
from compas.datastructures.mesh import Mesh
# or this
from compas.datastructures.mesh._mesh import Mesh
```

works for my VScode setup with Pylance

Just to make this more irritating, I do this with my other imports as well:
```
from compas.geometry._core._algebra import dot_vectors
from compas.geometry.primitives.frame import Frame
from compas.geometry.primitives.vector import Vector
from compas.robots.model.robot import RobotModel
```

**Comment by lottilotte** [comment]
Many thanks for the suggestion! Unfortunately for teaching this is rather unpractical...

**Comment by tomvanmele** [comment]
For me the `jedi` language server works perfectly. To activate, try the following steps and if that doesn't work we can do a zoom session...

1. open settings (the little gear wheel at the bottom left of the window + select settings)
2. search for "language server" on the settings page
3. in the the search results select "Python" in the category options
4. select `jedi` as the language server
5. reload the window to activate (open the command palette and select `Developer: Reload window`)

![Screenshot 2021-03-10 at 20 25 25](https://user-images.githubusercontent.com/29653489/110686293-94daf600-81df-11eb-94f8-285bf5c6bd1e.png)

**Comment by lottilotte** [comment]
Hi, Yes no this doesn’t work unfortunately… Many thanks for your help. What time suits you best for zoom?

**Comment by tomvanmele** [comment]
now?

**Comment by tomvanmele** [comment]
@duchaoyu @lottilotte please check pylance branch. problem should be solved...

**Comment by tomvanmele** [comment]
solved since `1.2.0`

**Comment by lottilotte** [comment]
Thank you!
